### PR TITLE
fs2-rabbit backend

### PR DESCRIPTION
**Pull request by andrewgee**

### CODE REVIEW

File: backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+class Fs2RabbitAmqpClient[F[_]](client: RabbitClient[F], connection: model.AMQPConnection, publishChannel: model.AMQPChannel)(implicit F: Async[F]) extends AmqpClient[F] {
+
+  implicit val deliveryEncoder: MessageEncoder[F, PublishCommand] =
+    Kleisli { publishCommand =>
+      val fs2MessageHeaders: Map[String, model.AmqpFieldValue] = publishCommand.basicProperties.headers.mapValues {
+        case bd: java.math.BigDecimal => DecimalVal.unsafeFrom(bd)
+        case ts: java.time.Instant    => TimestampVal.from(ts)
+        case d: java.util.Date        => TimestampVal.from(d)
+        //        case t: java.util.Map[String@unchecked, AnyRef@unchecked] =>
+        //          TableVal(t.asScala.toMap.map { case (key, v) => ShortString.unsafeFrom(key) -> unsafeFrom(v) })
```
Comment (thomasl8) [inline backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala:34]: To remove?

File: backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      case decl.ExchangeBinding(destinationExchangeName, sourceExchangeName, routingKey, arguments) => ???
+      case decl.Queue(name, isDurable, isExclusive, shouldAutoDelete, arguments) =>
+        client.declareQueue(DeclarationQueueConfig.default(model.QueueName(name.value)).copy(arguments = argumentsFromAnyRef(arguments)))
+    }
+
+  }
+
+  override def publisher(): Publisher[F, publish.PublishCommand] = (publishCommand: PublishCommand) => {
+    implicit val channel: model.AMQPChannel = publishChannel
+    println(publishCommand)
```
Comment (thomasl8) [inline backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala:132]: println

File: backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+  override def isConnectionOpen: F[Boolean] = F.pure(connection.value.isOpen)
+
+}
+
+object Fs2RabbitAmqpClient {
+  def apply[F[_]](config: AmqpClientConfig)(implicit async: Async[F]): Resource[F, AmqpClient[F]] = {
+    val fs2RabbitConfig = Fs2RabbitConfig(
+      config.host, config.port, "/", 10.seconds, ssl = false, Some(config.username), Some(config.password), requeueOnNack = false, requeueOnReject = false, Some(10)
```
Comment (thomasl8) [inline backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala:170]: I was thinking to get the `RequeueImmediate` working we could set `requeueOnReject` to true and use `reject` for `RequeueImmediate`.

File: example/src/main/scala/com/itv/bucky/example/requeue/Fs2RabbitAmqpClient.scala [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  ExchangeType => Fs2ExchangeType,
+  QueueName => Fs2RabbitQueueName,
+  RoutingKey => Fs2RoutingKey
+}
+import scodec.bits.ByteVector
+
+import java.nio.charset.StandardCharsets.UTF_8
+import scala.concurrent.duration.FiniteDuration
+
+class Fs2RabbitAmqpClient private (client: RabbitClient[IO], connection: AMQPConnection, publishChannel: AMQPChannel) extends AmqpClient[IO] {
```
Comment (thomasl8) [inline example/src/main/scala/com/itv/bucky/example/requeue/Fs2RabbitAmqpClient.scala:47]: Not needed in examples anymore

Comment (andrewgee) [reply example/src/main/scala/com/itv/bucky/example/requeue/Fs2RabbitAmqpClient.scala]: gone!

File: backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala (hunk 3 above)
Comment (andrewgee) [reply backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala]: great idea, done

**Review by thomasl8 — APPROVED** [review]

**Pull request merged**